lib/utils/getRetirement/index.ts

```diff
@@ -9,37 +9,48 @@ import {
 } from "../../utils";
 
 import {
-  Retirements,
+  RetirementTotals,
   RetirementsResult,
   RetirementIndexInfo,
   RetirementIndexInfoResult,
 } from "../../types/offset";
 
+export const createRetirementStorageContract = (
+  provider: providers.JsonRpcProvider
+) => {
+  return new ethers.Contract(
+    addresses["mainnet"].retirementStorage,
+    KlimaRetirementStorage.abi,
+    provider
+  );
```

Comment:
I wonder if in the future we should just instantiate and export every contract once, with a default provider, maybe from `@klimadao/lib`
and then when we use them for user transactions which need private key, we set the provider

```ts
import { retirementStorage } from "@klimadao/lib/contracts";

const doTransaction = (params) => {
  retirementStorage.provider = params.provider;
  const totals = retirementStorage.getRetirementTotals(params.address);
}
```

Reply:
Good idea, let´s keep that in mind.